File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int max = 0;
        int left = 0;
        int right = heights.size() - 1;

        while (left < right) {
            int mi = min(heights[left], heights[right]);
            int cur = (right - left) * mi;

            if (cur > max) max = cur;

            if (heights[left] < heights[right]) left++;
            else right--; 
        }
        return max;
    }
};

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-1.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        res = 0
        n = len(heights)

        for i in range(n):
            left = i
            while left >= 0 and heights[left] >= heights[i]:
                left-=1
            right = i
            while right < n and heights[right] >= heights[i]:
                right+=1
            
            left+=1
            right-=1
            area = (right - left + 1) * heights[i]
            res = max(area, res)
        
        return res
            